### PR TITLE
Unite app.nit metadata annotations common to Android and iOS

### PR DESCRIPTION
**Pull request by xymus**
* Rename the annotation `java_package` to `app_namespace`.
* Move up `app_name`, `app_version` and `app_namespace` and their default values from the Android platform.
* Apply these annotations in iOS too.
* Use the annotations in Hello iOS.

### CODE REVIEW

**Comment by privat** [comment]
+1. nice

**Comment by Morriar** [comment]
Useful meme

In doubt, do not read it anyway. See you tomorrow.

**Comment by xymus** [comment]
Don't worry, it was posted on march 31!

File: src/platform/app_annotations.nit [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import literal
+import semantize
+private import annotation
+
+# Metadata associated to an `app.nit` project
+class AppProject
+	# Pretty name of the resulting application
+	var name: String = mainmodule.first_real_mmodule.name is lazy
+
+	# Short project used in `namespace` and configuration files
```
Comment (Morriar) [inline src/platform/app_annotations.nit]: Short project *name* used in... ?

File: src/platform/app_annotations.nit [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+	# Namespace/package used to identify the application
+	var namespace = "org.nitlanguage.{short_name}" is lazy
+
+	# Version of the application
+	var version = "0.1"
+
+	# Numerical version code of th application
+	var version_code: Int is lazy do
+
```
Comment (Morriar) [inline src/platform/app_annotations.nit]: That blank line bothers me

File: src/platform/app_annotations.nit [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	private var modelbuilder: ModelBuilder
+	private var mainmodule: MModule
+
+	init
+	do
+		var annot = modelbuilder.lookup_annotation_on_modules("app_name", mainmodule)
+		if annot != null then
+			var val = annot.arg_as_string(modelbuilder)
+			if val != null then name = val
```
Comment (Morriar) [inline src/platform/app_annotations.nit]: So using that annotation I can set the `name` but it will not affect the `short_name`?

**Comment by Morriar** [comment]
+1

File: src/platform/app_annotations.nit (hunk 2 above)
Comment (xymus) [inline src/platform/app_annotations.nit]: Me too! but sadly the grammar does not support moving the `do` on the next line.

File: src/platform/app_annotations.nit (hunk 3 above)
Comment (xymus) [inline src/platform/app_annotations.nit]: Exactly, in theory the short_name is never displayed to the user or the developer.

**Comment by xymus** [comment]
Rerolled to fix errors in doc.

File: src/platform/app_annotations.nit (hunk 2 above)
Comment (Morriar) [inline src/platform/app_annotations.nit]: But can you remove the blank line?

Alexandre

2015-04-02 11:45 GMT-04:00 Alexis Laferrière <notifications@github.com>:

**Comment by xymus** [comment]
ok to test

**Pull request merged**